src/apps/lwaftr/fragmentv4.lua

```diff
@@ -128,3 +128,120 @@ function fragment_ipv4(ipv4_pkt, mtu)
 
    return FRAGMENT_OK, pkts
 end
+
+
+REASSEMBLE_OK = 1
+REASSEMBLE_INVALID = 2
+REASSEMBLE_MISSING_FRAGMENT = 3
+
+
+function reassemble_ipv4(fragments, l2_size)
+   local flags_and_frag_offset_offset = l2_size + constants.o_ipv4_flags
+   table.sort(fragments, function (pkt1, pkt2)
+       local pkt1_offset = band(C.ntohs(rd16(pkt1.data + flags_and_frag_offset_offset)),
+                                frag_offset_field_mask)
+       local pkt2_offset = band(C.ntohs(rd16(pkt2.data + flags_and_frag_offset_offset)),
+                                frag_offset_field_mask)
+       return pkt1_offset < pkt2_offset
+   end)
+
+   -- Check that first fragment has a 0 as fragment offset.
+   if band(C.ntohs(rd16(fragments[1].data + flags_and_frag_offset_offset)),
+           frag_offset_field_mask) ~= 0
+   then
+      return REASSEMBLE_MISSING_FRAGMENT
+   end
+
+   -- Check that the last fragment does not have "more fragments" flag set
+   if band(C.ntohs(rd16(fragments[#fragments].data + flags_and_frag_offset_offset)),
+           flag_more_fragments_mask) ~= 0
+   then
+      return REASSEMBLE_MISSING_FRAGMENT
+   end
+
+   local ihl = get_ihl_from_offset(fragments[1], l2_size)
+   local header_size = l2_size + ihl
+   local frag_id_offset = l2_size + constants.o_ipv4_identification
+   local frag_id = rd16(fragments[1].data + frag_id_offset)
+   local packet_size = fragments[1].length
+   local fragment_lengths = { packet_size - header_size }
+   local fragment_offsets = { 0 }
+   local status = REASSEMBLE_OK
+
+   for i = 2, #fragments do
+      local fragment = fragments[i]
```

Comment:
That was my first thought when implementing, but note that the first packet is handled differently and `ipairs()` does not allow to specify the start index of the iteration. The alternative would be to check for `i == 1` inside the loop body like this:

```lua
for i, fragment = ipairs(fragments) do
  if i == 1 then
    -- Handle first packet differently
  else
    -- Handle all the other packets
  end
end
```

So I think using a numeric `for`-loop is acceptable in this case, which avoids the extra check inside the loop.